### DISCUSSION TITLE
Loading xi:include xml files from the program code

### DISCUSSION THREAD

**Discussion by davisjamesf** [pelicanmapping/osgearth#2403] [Support Q&A] [answered]
I have been using the .earth file to organize annotations, features, labels etc. by storing them in different .xml files and loading them by using the <xi:include href="myData.xml"/> in the earth file.   

I'd like to dynamically load and unload these same xml files in my c++ code instead of the .earth file by sending in the file name/location in as a function argument and appending a node to the earth node (or whatever is appropriate to add the child to).

Is this possible, and are there any examples I can follow?  Thanks for your help.

**Comment by gwaldron** [comment, marked as answer]
I don't think there's a straight up example, but the steps would be something like:
* Use `XmlDocument::load()` to load the file
* Extract the `Config` object from the first `XmlElement` in the document by calling `getConfig()` on it
* Call `Layer::create(config)` to create your new map layer from the `Config`. 

You can look at the code that processes `<xi:include>` for guidance, near XmlUtils.cpp:253
Hope this helps :)

**Reply by davisjamesf** [reply]
It was Layer::create(config) that was the missing piece of my puzzle.  I'll try it out.  Thanks!

**Reply by davisjamesf** [reply]
Here is the (very unpolished) code that worked for me.  Hopefully it'll help someone else with the same question:

```
std::string layerfile = "myLayer.xml";
XmlDocument *doc = XmlDocument::load(layerfile);
XmlElement *el = static_cast<XmlElement*>(doc->getChildrent().front().get());
Config conf = static_cast<XmlElement*>(el)->getConfig(layerstr);
m_mapNode->getMap()->addLayer(Layer::create(conf));
```
